packages/geolocator/tizen/src/geolocator_tizen_plugin.cc

```diff
@@ -49,200 +143,117 @@ class GeolocatorTizenPlugin : public flutter::Plugin {
-                            tizen_result.message());
-          delete result_ptr;
-        });
   }
 
-  void OnGetLastKnownPosition(
-      std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result) {
-    Location location;
-    TizenResult tizen_result =
-        location_manager_->GetLastKnownLocation(&location);
-    if (!tizen_result) {
-      result->Error("Failed to get last known position.",
-                    tizen_result.message());
+  void OnRequestPermission() {
+    PermissionStatus result =
+        permission_manager_->RequestPermission(kPrivilegeLocation);
+    if (result == PermissionStatus::kDeniedForever ||
+        result == PermissionStatus::kDenied) {
+      SendErrorResult("Permission denied", "Permission denied by user.");
+      return;
+    } else if (result == PermissionStatus::kError) {
+      SendErrorResult("Operation failed", "Failed to request permission.");
       return;
     }
-    result->Success(location.ToEncodableValue());
+    SendResult(flutter::EncodableValue(static_cast<int>(result)));
   }
 
-  void OnGetCurrentPosition(
-      std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result) {
-    auto result_ptr = result.release();
-    TizenResult tizen_result = location_manager_->RequestCurrentLocationOnce(
-        [result_ptr](Location location) {
-          result_ptr->Success(location.ToEncodableValue());
-          delete result_ptr;
-        },
-        [result_ptr](TizenResult error) {
-          result_ptr->Error(
-              "An error occurred while requesting current location.",
-              error.message());
-          delete result_ptr;
-        });
-
-    if (!tizen_result) {
-      result_ptr->Error("Failed to call RequestCurrentLocationOnce.",
-                        tizen_result.message());
-      delete result_ptr;
+  void OnGetLastKnownPosition() {
+    try {
+      Position position = location_manager_->GetLastKnownPosition();
+      SendResult(position.ToEncodableValue());
+    } catch (const LocationManagerError &error) {
+      SendErrorResult("Operation failed", error.GetErrorString());
     }
   }
 
-  void OnListenGeolocatorServiceUpdates(
-      std::unique_ptr<flutter::EventSink<flutter::EncodableValue>>
-          &&event_sink) {
-    geolocator_service_updates_event_sink_ = std::move(event_sink);
-    TizenResult tizen_result = location_manager_->SetOnServiceStateChanged(
-        [this](ServiceState service_state) {
-          flutter::EncodableValue value(static_cast<int>(service_state));
-          geolocator_service_updates_event_sink_->Success(value);
-        });
-    if (!tizen_result) {
-      LOG_ERROR("Failed to set OnServiceStateChanged, %s.",
-                tizen_result.message().c_str());
-      geolocator_service_updates_event_sink_ = nullptr;
+  void OnGetCurrentPosition() {
+    auto result_ptr = result_.get();
+    try {
+      location_manager_->GetCurrentPosition(
```

Comment:
When the `GetCurrentPosition` is performed without errors, when is the `nullptr` assigned to `result_`?

What if another method request arrives before the callback is executed?

I think `GeolocatorTizenPlugin` should give up ownership of `result_` completely rather than pass reference of `result_` to the callback.

Reply:
if you capture this pointer in `lambda' for callback functions and still want to keep result as a member, I think you have to add some guard to prevent the result pointer from being replaced with a new one.

Let's assume the following situation.
 1. Register a callback for asynchronously processed requests, the plugin still has the `result` pointer that has not yet sent a result as a member.
 2. a new method request arrives to plugin, but the registered callback is still not executed.
 3. The result pointer is replaced by a new one.
 4. Finally the callback is executed, sending the result to the wrong place.

Reply:
I changed it to release() `result_` unique ptr.